packages/core/src/utils.js

```diff
@@ -767,6 +767,31 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     return resolveCondition(schema, rootSchema, formData);
   }
 
+  if (resolvedSchema.properties) {
+    const properties = {};
+
+    Object.entries(resolvedSchema.properties).forEach(entries => {
+      const propName = entries[0];
+      const propSchema = entries[1];
```

Comment:
Is it possible that a resolved schema will have additional properties such that this ordering doesn't hold? Seems like it would make more sense to filter

Reply:
Could you provide the schema which is not handled correctly by this code? For better understanding the case you are talking about. I use this code in the project with large and complex schemas with nested if/then/else so if you see the case which will not work - tell me please.

Reply:
I believe as we're using `entries` here, and the properties will always be objects representing the fields, index 0 will always be `propName` and index 1 the subschema ?

Reply:
Sure, but I guess @nickgros is talking about something like `.additionalProperties`.

Reply:
I see now, I think I was misunderstanding the code. This makes sense to me. Thanks for the clarification